File: tests/javascript/testB.js
class TestB {
    static printHelloTestB1() {
        console.log("Hello, World!");
        for (let i = 0; i < 5; i++) {
            console.log("This is line " + i);
            if (i % 2 === 0) {
                console.log("Even number");
            } else {
                console.log("Odd number");
            }
        }
    }

    static printHelloTestB2() {
        console.log("Hello, World!");
        for (let i = 0; i < 5; i++) {
            console.log("This is line " + i);
            if (i % 2 === 0) {
                console.log("Even number");
            } else {
                console.log("Odd number");
            }
        }
        console.log("End of printHello");
    }

    static printHelloTestB3() {
        console.log("Hello, World!");
        for (let i = 0; i < 5; i++) {
            console.log("This is line " + i);
            if (i % 2 === 0) {
                console.log("Even number");
            } else {
                console.log("Odd number");
            }
        }
    }

    static main() {
        this.printHelloTestB1();
        this.printHelloTestB2();
        this.printHelloTestB3();
    }
}

// Execute main method
TestB.main();

File: tests/rust/testB.rs
fn print_hello_test_b1() {
    println!("Hello, World!");
    for i in 0..5 {
        println!("This is line {}", i);
        if i % 2 == 0 {
            println!("Even number");
        } else {
            println!("Odd number");
        }
    }
}

fn print_hello_test_b2() {
    println!("Hello, World!");
    for i in 0..5 {
        println!("This is line {}", i);
        if i % 2 == 0 {
            println!("Even number");
        } else {
            println!("Odd number");
        }
    }
    println!("End of printHello");
}

fn print_hello_test_b3() {
    println!("Hello, World!");
    for i in 0..5 {
        println!("This is line {}", i);
        if i % 2 == 0 {
            println!("Even number");
        } else {
            println!("Odd number");
        }
    }
}

fn main() {
    print_hello_test_b1();
    print_hello_test_b2();
    print_hello_test_b3();
}